File: cpp/3rd_Week/Praktikum/latihanpraktikum3_cpp/Ram.cpp
#include <iostream>
#include <string>

using namespace std;

class Ram
{
private:
    /* data */
    string ramCapacity;
    long ramPrice;
public:
    Ram() {}
    Ram(string capacity, long price) {
        this->ramCapacity = capacity;
        this->ramPrice = price;
    }

    void setRamCapacity(string c){ this->ramCapacity = c; }
    void setRamPrice(long p){ this->ramPrice = p; }

    string getRamCapacity(){ return this->ramCapacity; }
    long getRamPrice(){ return this->ramPrice; }

    /* method print ram */
    void printRam(){
        cout << "-Data RAM" << endl;
        cout << " RAM Capicity : " << this->ramCapacity << endl;
        cout << " RAM Price    : " << this->ramPrice << endl;
        cout << "-----------------------------------" << endl;
    }

    ~Ram() {}
};

File: php/3rd_week/latihanpraktikum3_php/Ram.php
<?php

class Ram{
    // implementasi class Ram

    // atribut
    private $ramCapacity = "";
    private $ramPrice = 0;

    //constructor
    function __construct()
    {
    }

    //set and get atribut
    function setRamCapacity($capacity){
        $this->ramCapacity = $capacity;
    }

    function setRamPrice($price){
        $this->ramPrice = $price;
    }

    function getRamCapacity(){
        return $this->ramCapacity;
    }

    function getramPrice(){
        return $this->ramPrice;
    }

    // method print ram
    function printRam(){
        echo "-Data RAM". "<br/>";
        echo " Ram Capacity : ". $this->ramCapacity. "<br/>";
        echo " Harga RAM : ". $this->ramPrice. "<br/>";
        echo "--------------------------------------------". "<br/>";
    }

    //destructor
    function __destruct()
    {
    }
}
?>